pkg/cluster/kafkaservice/kafkaservice.go

```diff
@@ -0,0 +1,65 @@
+package kafkaservice
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/redhat-developer/app-services-cli/pkg/api/kas"
+	"github.com/redhat-developer/app-services-cli/pkg/cluster"
+	"github.com/redhat-developer/app-services-cli/pkg/cluster/constants/kafka"
+	"github.com/redhat-developer/app-services-cli/pkg/cluster/utils"
+	"github.com/redhat-developer/app-services-cli/pkg/kafka/kafkaerr"
+)
+
+type KafkaConnection struct {
+}
+
+func (k *KafkaConnection) CustomResourceExists(ctx context.Context, c *cluster.KubernetesCluster, namespace string, serviceName string, opts cluster.Options) error {
+
+	path := kafka.GetKafkaConnectionsAPIURL(namespace)
+
+	err := utils.ResourceExists(ctx, c, path, serviceName, opts)
+
+	return err
+}
+
+func (k *KafkaConnection) CreateCustomResource(ctx context.Context, c *cluster.KubernetesCluster, serviceID string, namespace string, opts cluster.Options) error {
```

Comment:
I feel that an options object is not needed here. It is very implementation specific. For example, what if I want to create a `XConnection` and not use `Localizer`?

My vote is to remove use of `options` from any interface methods, they can be passes as properties of the concrete type (KafkaConnection).

Reply:
Very very good point that follows good patterns!